create sha256 checksum files of tars and images

HiassofT opened the pull request:
@islipfd19 is this the missing functionality you are looking for?

islipfd19 commented:
@hiasoft -Yes, but I want to confirm. Will the sha256 file include only the filename or will it include the full path from where it resides? The sha256 will be placed in the target directory?

HiassofT commented:
yes, .sha256 files are created in the same directory as the tar etc files and only contain the filename without a path:

```
hias@tv:/scratch/libreelec/target$ ls -la LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5*
-rw-r--r-- 1 hias hias 126811281 Apr 24 14:04 LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.img.gz
-rw-r--r-- 1 hias hias       136 Apr 24 14:05 LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.img.gz.sha256
-rw-r--r-- 1 hias hias   7074760 Apr 24 14:04 LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.kernel
-rw-r--r-- 1 hias hias 128552960 Apr 24 14:04 LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.system
-rw-r--r-- 1 hias hias 140697600 Apr 24 14:04 LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.tar
-rw-r--r-- 1 hias hias       133 Apr 24 14:04 LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.tar.sha256
hias@tv:/scratch/libreelec/target$ cat LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.img.gz.sha256
7d9f6fa9a9de9c7d4db4b5ad8e4e012feacdb04c873a44363d300a958ef3236f  LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.img.gz
hias@tv:/scratch/libreelec/target$ cat LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.tar.sha256
9b2cc74e2c73779d8c8d0dc0822cf0860139c19335b25e4b8c0fe12a56ec8370  LibreELEC-RPi2.arm-9.0-devel-20180424140301-r28529-g180c0c68b5.tar
```

I've just updated scripts/image to also create .sha256 for wetek .zip files and fixed up the commit message of scripts/mkimage

MilhouseVH commented on scripts/mkimage:
```diff
@@ -327,6 +327,10 @@ fi # bootloader
+    # create sha256 checksum of ova image
+    ( cd $TARGET_IMG
+      sha256sum $(basename ${DISK/img/ova}) > $(basename ${DISK/img/ova}).sha256
+    )
```
It's not new to this PR, but the `${DISK/img/ova}` substitution looks rather fragile - I know it's designed to replace the `.img` extension but if `img` appears anywhere in `$DISK` (which is based on `$TARGET_IMG/$IMAGE_NAME.img `, ie. an absolute path) then it will fail - for instance `/home/jimgarner/LibreELEC.tv/target/LibreELEC-RPi2.arm-9.0-devel-20180424011910-r12345-g6ce3be6.img ` will fail.

@Hiassoft can I suggest we make this a little more robust? I think adding DISK_TMP, DISK_OVA, DISK_OVF and DISK_VMDK variables where `DISK` is initialised in line #46 would allow all the DISK substitutions to be removed entirely, and be much safer/reliable. Or add a "bare" DISK_NOEXT variable without the `.img` extension, so the appropriate extension can be added where required. Given the number of substitutions (14 or so) this may be better done in a separate PR.

HiassofT replied:
agreed, this is terrible and needs changing. I'll look at it

HiassofT commented:
I've refactored the DISK suffix substitution in mkimage, noticed I could use IMAGE_NAME instead of basename $DISK ...

Couldn't think of a better name than DISK_BASENAME (which is only half-way correct, as it's not a base name but a base path) but at least it should work.

Some thorough review of this refactoring would be highly appreciated, can't currently runtime test the ova.

HiassofT commented:
ah,sorry, just noticed I must have missed a part of your comment. If you prefer I can move the DISK replacement commit out and only add DISK_BASE/DISK_BASENAME for the sha here
